src/main/kotlin/Main.kt

```diff
@@ -74,27 +78,116 @@ fun main() = application {
                 ComponentScope(
                     onColorPicked = { colorName, color ->
                         val updatedColorPalette = updateColorPalette(
-                                currentColorPalette = currentColorPalette,
-                                colorName = colorName,
-                                color = color,
-                            )
+                            currentColorPalette = currentColorPalette,
+                            colorName = colorName,
+                            color = color,
+                        )
 
                         if (darkmode) {
                             darkColorScheme = updatedColorPalette
                         } else {
                             lightColorScheme = updatedColorPalette
                         }
+                    },
+                    onPrintColors = {
+                        shouldShowFileChooser = true
                     }
                 )
+
+                if (shouldShowFileChooser) {
+                    FileChooserDialog(
+                        title = "Select directory to save file",
+                        onResult = {
+                            saveColorsToFile(
+                                currentColorPalette, file = it, onResult = { success, message ->  
+                                   // Implement success / failure screens for this when VM is implemented,
+                                   // and we can easier manage states.
+                                }
+                            )
+                            shouldShowFileChooser = false
+                        },
+                        onCancel = {
+                            shouldShowFileChooser = false
+                        }
+                    )
+                }
             }
         }
     }
 }
 
+@Composable
+private fun FileChooserDialog(
+    title: String,
+    onResult: (result: File) -> Unit,
+    onCancel: () -> Unit
+) {
+    val fileChooser = JFileChooser(FileSystemView.getFileSystemView())
+    fileChooser.currentDirectory = File(System.getProperty("user.dir"))
+    fileChooser.dialogTitle = title
+    fileChooser.fileSelectionMode = JFileChooser.DIRECTORIES_ONLY
+    fileChooser.isAcceptAllFileFilterUsed = true
+    fileChooser.selectedFile = null
+    fileChooser.currentDirectory = null
+    if (fileChooser.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+        val file = fileChooser.selectedFile
+        onResult(file)
+    } else {
+        onCancel.invoke()
+    }
+}
+
+private fun saveColorsToFile(
+    currentColorPalette: ColorScheme,
+    file: File,
+    onResult: (success: Boolean, message: String) -> Unit
+) {
+    val path = file.path
+    val outputFile = Path(path, "colors.txt").createFile().toFile()
+    try {
+        FileOutputStream(outputFile, true).bufferedWriter().use { fileWriter ->
+            fileWriter.write("Exported Material 3 Colors" + "\n")
```

Comment:
I would export as a proper .kt file.

Reply:
also use kotlin `.apply`

Reply:
Good idea! Updated in [2fe7d83](https://github.com/oas004/Material3Playground/pull/19/commits/2fe7d83a53203a8f400ee3edf8e8965d9b601a4c) :) Thanks!